### PR TITLE
[MIG][13.0] stock_orderpoint_route: Migration to 13.0

### PR DESCRIPTION
**Pull request by guewen**
Port #785, #537

### CODE REVIEW

File: stock_orderpoint_route/models/stock_warehouse_orderpoint.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        domain="[('id', 'in', route_ids)]",
+        ondelete="restrict",
+    )
+
+    @api.depends("product_id", "warehouse_id", "warehouse_id.route_ids", "location_id")
+    def _compute_route_ids(self):
+        route_obj = self.env["stock.location.route"]
+        for wh in self.mapped("warehouse_id"):
+            wh_routes = wh.route_ids
+            for record in self.filtered(lambda r: r.warehouse_id == wh):
```
Comment (simahawk) [inline stock_orderpoint_route/models/stock_warehouse_orderpoint.py]: can we avoid double loop?
can we avoid browsing?

File: stock_orderpoint_route/models/stock_warehouse_orderpoint.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                routes = route_obj.browse()
+                if record.product_id:
+                    routes += record.product_id.mapped(
+                        "route_ids"
+                    ) | record.product_id.mapped("categ_id").mapped("total_route_ids")
+                if record.warehouse_id:
+                    routes |= wh_routes
+                parents = record.get_parents()
+                record.route_ids = routes.filtered(
```
Comment (simahawk) [inline stock_orderpoint_route/models/stock_warehouse_orderpoint.py]: can we split this to `_get_routes_whatever`?

File: stock_orderpoint_route/models/stock_warehouse_orderpoint.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    lambda route: any(
+                        p.location_id in parents
+                        for p in route.rule_ids.filtered(
+                            lambda rule: rule.action in ("pull", "pull_push")
+                        ).mapped("location_src_id")
+                    )
+                )
+
+    def get_parents(self):
```
Comment (simahawk) [inline stock_orderpoint_route/models/stock_warehouse_orderpoint.py:45]: can't this be solved using parent_path?

File: stock_orderpoint_route/tests/test_stock_orderpoint_route.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from odoo.tests import common
+
+
+class TestStockOrderpointRoute(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
```
Comment (simahawk) [inline stock_orderpoint_route/tests/test_stock_orderpoint_route.py:12]: set tracking disable pls

File: stock_orderpoint_route/tests/test_stock_orderpoint_route.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                "warehouse_id": cls.warehouse.id,
+                "company_id": cls.main_company.id,
+            }
+        )
+
+    @classmethod
+    def _create_user(cls, name, group_ids, company_ids):
+        return (
+            cls.env["res.users"]
+            .with_context({"no_reset_password": True})
```
Comment (simahawk) [inline stock_orderpoint_route/tests/test_stock_orderpoint_route.py]: use kw arg to not override ctx completely

**Review by simahawk — CHANGES_REQUESTED** [review]

File: stock_orderpoint_route/models/stock_warehouse_orderpoint.py (hunk 1 above)
Comment (guewen) [reply stock_orderpoint_route/models/stock_warehouse_orderpoint.py]: I ported the module as is, note: the number of warehouses is generally very low, shouldn't have too much an impact

File: stock_orderpoint_route/tests/test_stock_orderpoint_route.py (hunk 5 above)
Comment (guewen) [inline stock_orderpoint_route/tests/test_stock_orderpoint_route.py]:
```suggestion
            .with_context(no_reset_password=True)
```

File: stock_orderpoint_route/models/stock_warehouse_orderpoint.py (hunk 1 above)
Comment (guewen) [reply stock_orderpoint_route/models/stock_warehouse_orderpoint.py]: Fixed, we have less queries with a single, simple loop.

**Comment by guewen** [comment]
@simahawk rebased, fixed your comments in the last commit

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@JordiBForgeFlow could you pass by for a review? :)

**Review by jgrandguillaume — APPROVED** [review]
Functional review is ok on my side

**Pull request merged**